src/core/python/logger.cpp

```diff
@@ -29,18 +35,6 @@ static void PyLog(mitsuba::LogLevel level, const std::string &msg) {
     if (!name.empty() && name[0] != '<')
         fmt.insert(2, "()");
 
-    if (!Thread::thread()->logger()) {
```

Comment:
Maybe this could be replaced with an assert?

Reply:
I am not against that. My goal was to make it consistent with the C++ side here, which also just ignores this case:
https://github.com/mitsuba-renderer/mitsuba3/blob/master/include/mitsuba/core/logger.h#L140

Reply:
Understandable.. I guess it will crash  either way :D

Reply:
It should just ignore and not log anything at all, which is not too bad.

Reply:
Ahh sorry I didn't pay attention to the check you added above. Makes sense.

The case of `!Thread::thread()->logger()` should no longer happen by default now that you automatically bring over the main thread's logger, right?
If there's no logger, then it must be that the user set a thread environment that explicitly doesn't have a logger?

Reply:
Yes exactly. Not sure if that's something one would ever want?

In any case, there are still some other issues to be fixed with this PR. The current version does not shut down cleanly..